diff -Naur cwipi-0.8.2.old/src/fvm/fvmc_file.c cwipi-0.8.2/src/fvm/fvmc_file.c
--- cwipi-0.8.2.old/src/fvm/fvmc_file.c	2020-11-03 17:19:14.524730710 +0000
+++ cwipi-0.8.2/src/fvm/fvmc_file.c	2020-11-03 17:19:55.442155191 +0000
@@ -644,7 +644,7 @@
   lengths[0] = (global_num_end - global_num_start) * size;
   disps[0] = (global_num_start - 1) * size;
 
-  MPI_Type_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
+  MPI_Type_create_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
   MPI_Type_commit(&file_type);
 
   MPI_File_set_view(f->fh, f->offset, MPI_BYTE, file_type, datarep, f->info);
@@ -745,7 +745,7 @@
   lengths[0] = (global_num_end - global_num_start) * size;
   disps[0] = (global_num_start - 1) * size;
 
-  MPI_Type_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
+  MPI_Type_create_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
   MPI_Type_commit(&file_type);
 
   MPI_File_set_view(f->fh, f->offset, MPI_BYTE, file_type, datarep, f->info);
@@ -1073,7 +1073,7 @@
         char datarep[] = "native";
         lengths[0] = ni * size;
         disps[0] = 0;
-        MPI_Type_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
+        MPI_Type_create_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
         MPI_Type_commit(&file_type);
         MPI_File_set_view(f->fh, f->offset, MPI_BYTE, file_type,
                           datarep, f->info);
@@ -1183,7 +1183,7 @@
       char datarep[] = "native";
       lengths[0] = ni * size;
       disps[0] = 0;
-      MPI_Type_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
+      MPI_Type_create_hindexed(1, lengths, disps, MPI_BYTE, &file_type);
       MPI_Type_commit(&file_type);
       MPI_File_set_view(f->fh, f->offset, MPI_BYTE, file_type,
                         datarep, f->info);
diff -Naur cwipi-0.8.2.old/src/fvm/fvmc_locator.c cwipi-0.8.2/src/fvm/fvmc_locator.c
--- cwipi-0.8.2.old/src/fvm/fvmc_locator.c	2020-11-03 17:21:48.738099209 +0000
+++ cwipi-0.8.2/src/fvm/fvmc_locator.c	2020-11-03 17:26:11.187234306 +0000
@@ -1360,7 +1360,7 @@
 
   double comm_timing[4] = {0., 0., 0., 0.};
 
-  MPI_Aint extent;
+  MPI_Aint lb, extent;
 
   /* Get a free  request */
 
@@ -1376,7 +1376,7 @@
 
   /* Check extent of datatype */
 
-  MPI_Type_extent(datatype, &extent);
+  MPI_Type_get_extent(datatype, &lb, &extent);
   MPI_Type_size(datatype, &size);
 
   if (extent != size)
@@ -1512,7 +1512,7 @@
 
   double comm_timing[4] = {0., 0., 0., 0.};
 
-  MPI_Aint extent;
+  MPI_Aint lb, extent;
 
   /* Get a free  request */
 
@@ -1530,7 +1530,7 @@
 
   /* Check extent of datatype */
 
-  MPI_Type_extent(datatype, &extent);
+  MPI_Type_get_extent(datatype, &lb, &extent);
   MPI_Type_size(datatype, &size);
 
   nblocking_recv->size = size;
@@ -1640,12 +1640,12 @@
 
   double comm_timing[4] = {0., 0., 0., 0.};
 
-  MPI_Aint extent;
+  MPI_Aint lb, extent;
   MPI_Status status;
 
   /* Check extent of datatype */
 
-  MPI_Type_extent(datatype, &extent);
+  MPI_Type_get_extent(datatype, &lb, &extent);
   MPI_Type_size(datatype, &size);
 
   if (extent != size)
