diff -rNau cwipi-0.8.2.old/src/geomUtilities.cxx cwipi-0.8.2/src/geomUtilities.cxx
--- cwipi-0.8.2.old/src/geomUtilities.cxx	2014-12-10 15:16:36.000000000 +0000
+++ cwipi-0.8.2/src/geomUtilities.cxx	2020-11-04 10:28:49.372555332 +0000
@@ -20,6 +20,7 @@
 #include <algorithm>
 #include <vector>
 #include <cstdio>
+#include <cstdlib>
 
 #include <bftc_printf.h>
 
@@ -626,7 +627,7 @@
       int nPolyhedraVertices = 0;
       for (int iface = 0; iface < nPolyFace; iface++) {
 
-        const int face          = abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
+        const int face          = std::abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
         const int faceIdx       = faceConnectivityIdx[face];
         const int nFaceVertices = faceConnectivityIdx[face+1] - faceIdx;
 
@@ -677,7 +678,7 @@
 
       for (int iface = 0; iface < nPolyFace; iface++) {
 
-        const int face          = abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
+        const int face          = std::abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
         const int direction     = (cellToFaceConnectivity[polyIdx + iface] < 0) ? -1 : 1;
 
         const int faceIdx       = faceConnectivityIdx[face];
@@ -772,7 +773,7 @@
 
       for (int iface = 0; iface < nPolyFace; iface++) {
 
-        const int face          = abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
+        const int face          = std::abs(cellToFaceConnectivity[polyIdx + iface]) - 1;
         const int direction     = (cellToFaceConnectivity[polyIdx + iface] < 0) ? -1 : 1;
         const double*surfaceVector_f = surfaceVector + 3*face;
 
